src/components/action-popover/action-popover.spec.tsx

```diff
@@ -1519,36 +2002,353 @@ describe("ActionPopover", () => {
     });
   });
 
-  describe("when the horizontalAlignment prop is set to right", () => {
-    beforeEach(() => {
-      wrapper = enzymeMount(
-        <ThemeProvider theme={mintTheme}>
-          <ActionPopover horizontalAlignment="right">
-            <ActionPopoverItem key="1" href="#" icon="download">
-              test download
-            </ActionPopoverItem>
-          </ActionPopover>
-        </ThemeProvider>
-      );
-    });
+  describe("padding checks on 'StyledMenuItemInnerText'", () => {
+    it.each([
+      ["left", "left"],
+      ["left", "right"],
+      ["right", "left"],
+      ["right", "right"],
+    ])(
+      "when horizontalAlignment is %s, submenuPosition is % and iconState, caretState, icon and submenu are all false, left and right padding is: var(--spacing100)",
+      (alignment, position) => {
+        renderDesignVariants(1, {
+          submenuPosition: position,
+          horizontalAlignment: alignment,
+        });
+        openMenu();
+        assertStyleMatch(
+          {
+            paddingLeft: "var(--spacing100)",
+            paddingRight: "var(--spacing100)",
+          },
+          wrapper.find(StyledMenuItemInnerText).at(0)
+        );
+      }
+    );
 
-    it("then menu item content should be right aligned", () => {
-      openMenu();
-      assertStyleMatch(
-        {
-          justifyContent: "flex-end",
-        },
-        wrapper.find(StyledMenuItem)
-      );
-    });
+    it.each([
+      ["left", "left", "var(--spacing400)", "var(--spacing100)"],
+      ["left", "right", "var(--spacing100)", "var(--spacing100)"],
+      ["right", "left", "var(--spacing100)", "var(--spacing100)"],
+      ["right", "right", "var(--spacing100)", "var(--spacing400)"],
+    ])(
+      "when horizontalAlignment is %s, submenuPosition is %s and iconState is false, caretState is true, icon is false and submenu is false padding-left is: %s and padding-right is: %s",
```

Comment:
**comment**: I think similar comments to those I already made on the new Cypress tests apply to these new unit tests too.

Further (and as I hadn't realised when first commenting on the Cypress tests), we can hopefully get rid of a lot of these if my comments about being able to remove `iconState` etc as props turn out to not be crazy 😅

Reply:
I agree about the descriptions being unreadable, but see my comment explaining my view on the state props. Happy to work out what to do going forward 👍